File: 1635-number-of-good-pairs/1635-number-of-good-pairs.cpp
class Solution {
public:

    int numIdenticalPairs(vector<int>& nums) {
        map<int, int> mp;
        int sum(0);
        for(int i : nums) {
            mp[i] ++;
        }

        for (std::map<int, int>::iterator it = mp.begin(); it != mp.end(); ++it) {
           if(it->second>1) {
            int n = it->second;
            sum += (n*(n-1)/2);
           }
       }

       return sum;

    }
};

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.py
class Solution(object):
    def numIdenticalPairs(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d = {}
        for i in nums:
            d[i] = d.get(i, 0) + 1
        s=0
        for i in d.values():
            if i > 1:
                n = i
                s += n*(n-1)/2
        return s

            